### ISSUE TITLE
Refactoring proposal for logging

### ISSUE THREAD

**Issue by sangkeun00** [logix-project/logix#65]
In essence, many states that AnaLog tracks for now are essentially covariance. For example, KFAC Hessian is basically (un-centered) covariance of forward activations and backward error signals, and raw Hessian is simply (un-centered) covariance of gradient. EK-FAC also updates eigenvalues based on per-sample gradient from training data. Therefore, I believe it's safe to say that the main features of logging in AnaLog are:

1. Extract relevant logs (e.g., forward, backward, gradient) from neural networks using hooks
2. Compute some statistic (e.g., mean, covariance) using these logs

### Problem

However, at the moment, our code is structured around the whole algorithm (e.g. KFAC) instead of basic operations (e.g. mean, covariance). For instance, `KFACHessianHandler` implements both covariance computation and EKFAC update mechanisms. In addition, RawHessianHandler implements the covariance computation mechanism separately from KFACHessianHandler. In the end, our current code lacks the modularity, and I believe this would negatively affect the maintenance of our repo in the future, especially as we add more and more features.

### Proposed solution

To address this issue, I propose to refactor our code base around the basic operations. For example, our code for logging KFAC/EKFAC can be re-written as below:

```python
from analog.log import Mean, Covariance, Log, Save, EKFAC

# KFAC
analog.update_logging_config(
    "forward": [Covariance], "backward": [Covariance], "grad": [Save]
)

# EKFAC
analog.update_logging_config(
    "forward": [], "backward": [], "grad": [EKFAC]
)

# eval
analog.update_logging_config(
    "forward": [], "backward": [], "grad": [Log]
)
```

This way, we can improve the modularity of our code, while also allowing researchers to implement their new algorithms with AnaLog. For users who don't know how to set these `logging_config`, we will use `AnaLogScheduler` to automatically set this config based on some template.

@hwijeen @pomonam @hage1005 @YoungseogChung
